opentracing-util/src/main/java/io/opentracing/util/GlobalTracer.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.util;
+
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+
+import java.util.Iterator;
+import java.util.ServiceLoader;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Forwards all methods to another tracer that can be configured in one of two ways:
+ * <ol>
+ * <li>Explicitly, calling {@link #register(Tracer)} with a configured tracer, or:</li>
+ * <li>Automatically using the Java {@link ServiceLoader} SPI mechanism to load a {@link Tracer} from the classpath.</li>
+ * </ol>
+ * <p>
+ * When the tracer is needed it is lazily looked up using the following rules:
```

Comment:
My two cents is that GlobalTracer should aspire to be nothing more than a global Tracer. All of the concerns about wrapping, etc, can be handled by layers above the core OT API... the only reason I want GlobalTracer to be part of OT proper is that we do want a global/singleton Tracer *accessor* if we expect library instrumentation code to be easy to write and maintain.

I am a little more sympathetic about the ServiceLoader functionality (than the wrapper), but still feel like it doesn't *need* to be here to satisfy the core requirement of library instrumentation, and thus can be left out for now. Adding it in later will increase the line count but shouldn't change the API to access the global Tracer.

I also think it is fine to make the GlobalTracer "write-once"... i.e., no `update(...)` functionality. As was mentioned previously, even if we support that in the GlobalTracer, it could wreak havoc with Tracers/Spans that assume they know the types of other Spans and SpanContexts propagating through the process at the time.

PS: I'm sorry to take so long to follow up on this – my calendar is insane right now and I'm falling behind on ~everything.